src/caffe/solver.cpp

```diff
@@ -435,6 +435,26 @@ void Solver<Dtype>::Snapshot() {
 }
 
 template <typename Dtype>
+void Solver<Dtype>::CheckSnapshotWritePermissions() {
+  if (param_.snapshot()) {
+    if (!param_.has_snapshot_prefix()) {
+      LOG(FATAL) << "In solver params, snapshot is specified "
+        << "but snapshot_prefix is not";
+    }
```

Comment:
A minor issue: better to use `CHECK` marcos here and below to make code more succinct.